tests/test_set_container_iterator.cpp

```diff
@@ -0,0 +1,247 @@
+#include <catch2/catch.hpp>
+#include <ftl/ftl.hpp>
+#include <vector>
+#include <list>
+#include <utility>
+
+#define TEST_TAG "[set_container_iterator]"
+
+TEST_CASE(TEST_TAG "collect to std::vector", TEST_TAG) {
+  ftl::set<std::string> set = { { "red", "green", "blue" } };
+
+  auto mapped_set = set.iter().collect<std::vector<std::string>>();
+
+  REQUIRE(mapped_set == std::vector<std::string>{ "blue", "green", "red" });
+}
+
+TEST_CASE(TEST_TAG "collect const to std::vector", TEST_TAG) {
+  const ftl::set<int> set = { { 1, 2, 3 } };
+
+  auto mapped_set = set.iter().collect<std::vector<int>>();
+
+  REQUIRE(mapped_set == std::vector<int>{ { 1, 2, 3 } });
+}
+
+TEST_CASE(TEST_TAG "collect to std::set", TEST_TAG) {
+  ftl::set<int> set = { { 1, 2, 3 } };
+
+  auto mapped_set = set.iter().collect<std::set<int>>();
+
+  REQUIRE(mapped_set == std::set<int>{ { 1, 2, 3 } });
+}
+
+TEST_CASE(TEST_TAG "collect const to std::set", TEST_TAG) {
+  const ftl::set<int> set = { { 1, 2, 3 } };
+
+  auto mapped_set = set.iter().collect<std::set<int>>();
+
+  REQUIRE(mapped_set == std::set<int>{ { 1, 2, 3 } });
+}
+
+TEST_CASE(TEST_TAG "set collect", TEST_TAG) {
```

Comment:
Fix test naming to follow convention. This one should be "map collect", because these are the 2 operations being tested. Word "set" is redundant. Do this in every other test that doesn't match convetion.